hack/generator/pkg/codegen/storage/storage_type_factory.go

```diff
@@ -0,0 +1,208 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package storage
+
+import (
+	"github.com/Azure/azure-service-operator/hack/generator/pkg/astmodel"
+	"github.com/pkg/errors"
+	"k8s.io/klog/v2"
+)
+
+// StorageTypeFactory is used to create storage types for a specific service
+type StorageTypeFactory struct {
+	service                    string                                                  // Name of the group we're handling (used mostly for logging)
+	types                      astmodel.Types                                          // All the types for this group
+	pendingStorageConversion   astmodel.TypeNameQueue                                  // Queue of types that need storage variants created for them
+	pendingConversionInjection astmodel.TypeNameQueue                                  // Queue of types that need conversion functions injected
+	pendingMarkAsHubVersion    astmodel.TypeNameQueue                                  // Queue of types that need to be flagged as the hub storage version
+	idFactory                  astmodel.IdentifierFactory                              // Factory for creating identifiers
```

Comment:
minor, style: Is the description in a tabbed-over string a standard golang style? (@babbageclunk knows maybe?)

It reads pretty cleanly here, but also isn't a "golang-style" comment as it's not starting with the name of the variable it refers to?

Mostly just wondering for my own knowledge.

Reply:
I'm sure I've seen this - can change it if not idiomatic